packages/styled-components/src/utils/flatten.js

```diff
@@ -14,19 +14,32 @@ import addUnitIfNeeded from './addUnitIfNeeded';
  */
 const isFalsish = chunk => chunk === undefined || chunk === null || chunk === false || chunk === '';
 
-export const objToCss = (obj: Object, prevKey?: string): string => {
-  const css = Object.keys(obj)
-    .filter(key => !isFalsish(obj[key]))
-    .map(key => {
-      if (isPlainObject(obj[key])) return objToCss(obj[key], key);
-      return `${hyphenate(key)}: ${addUnitIfNeeded(key, obj[key])};`;
+
+const objToCssArray = (obj: Object, prevKey?: string): Array<string | Function> => {
+  const rules = []
+  const keys = Object.keys(obj)
+
+  keys
```

Comment:
this would be cleaner as a reduce I think

Reply:
true, but usually it's slower than `.forEach` because the result has different type than the items. i did a quick benchmark for that [here](http://jsben.ch/A6Ad9)

Reply:
Fair enough!